src/components/kepler-gl.js

```diff
@@ -138,30 +140,47 @@ export const mapFieldsSelector = props => ({
   locale: props.uiState.locale
 });
 
-export const sidePanelSelector = (props, availableProviders) => ({
-  appName: props.appName,
-  version: props.version,
-  appWebsite: props.appWebsite,
-  mapStyle: props.mapStyle,
-  onSaveMap: props.onSaveMap,
-  uiState: props.uiState,
-  mapStyleActions: props.mapStyleActions,
-  visStateActions: props.visStateActions,
-  uiStateActions: props.uiStateActions,
+export function getVisibleDatasets(datasets) {
+  // We don't want Geocoder dataset to be present in SidePanel dataset list
+  return filterObjectByPredicate(datasets, (key, value) => key !== GEOCODER_DATASET_NAME);
+}
 
-  datasets: props.visState.datasets,
-  filters: props.visState.filters,
-  layers: props.visState.layers,
-  layerOrder: props.visState.layerOrder,
-  layerClasses: props.visState.layerClasses,
-  interactionConfig: props.visState.interactionConfig,
-  mapInfo: props.visState.mapInfo,
-  layerBlending: props.visState.layerBlending,
+export const sidePanelSelector = (props, availableProviders) => {
+  // visibleDatasets
+  const filteredDatasets = getVisibleDatasets(props.visState.datasets);
 
-  width: props.sidePanelWidth,
-  availableProviders,
-  mapSaved: props.providerState.mapSaved
-});
+  // And we don't want Geocoder dataset to be present in the tooltip field picker
+  const interactionConfigClone = cloneDeep(props.visState.interactionConfig);
```

Comment:
don't use `cloneDeep` in compoennt.

Do this filter in the interaction config panel, instead of here

Reply:
```js
        {Object.keys(config.fieldsToShow).map(dataId => 
           dataId === GEOCODER_DATASET_NAME ? null : 
          (<DatasetTooltipConfig
            key={dataId}
            config={config}
            onChange={onChange}
            dataset={datasets[dataId]}
          />)
        ))}
```

Reply:
Thanks, will do!

Reply:
cloneDeep shouldn't be used inside a component because it is slow or is there some other reason?

Reply:
it is really slow since he copied every single property recursively

Reply:
Yeah, that's what I thought. New commit in a minute.

Reply:
Also, react does shallow compare on props to determine whether a component should be rerendered. deep clone will cause all the components w/ `interactionConfig` as props to rerender all the time

Reply:
Fixed.